### ISSUE TITLE
LuaErrorKind::IncompleteStatement should be removed.

### ISSUE THREAD

**Issue by calviken** [chucklefish/rlua#2]
As seen [here](https://github.com/chucklefish/rlua/blob/5c0f183a52ec65bf0c5422cc849138a457e0a5eb/src/util.rs#L124), rlua checks for "\<eof>" at the end of error messages and returns LuaErrorKind::IncompleteStatement if it finds it.

One issue I have with this is that it causes tests::test_eval to fail when using LuaJIT, because the error message ends with "'\<eof>'" (notice the single quotes). I realize that might not be a good enough reason to change it.

But my OTHER issue is that it just seems pointless to have it at all. Why not just return a LuaErrorKind::ScriptError in both cases? (the comment there even suggests that it would be better to remove it)

**Comment by calviken** [comment]
Update: After giving it a minimal amount of thought, I've decided that this whole LuaJIT integration project of mine is actually a terrible idea, so go ahead and disregard everything I've said about that.

I think my second point still stands though.

**Comment by kyren** [comment]
The original reason to have it was to be able to reimplement something similar to the lua repl, it's kind of neat to be able to start a lua repl which in reality is your "game environment", it can even allow you to do repl-like testing in a language with no repl.

You handle IncompleteStatement as a separate kind of error, and instead of erroring you just print >>, like the lua repl.

Now, I agree HOW the lua repl detects EOF is pretty ridiculous.  I mean, I shouldn't judge it works for them, but it's at least weird for our purposes.

I can't make any promises, but it's pretty clear people want LuaJIT, and I'll.. *try* to look into how difficult it would be in general.  I really can't promise anything though, because it's just so low priority for me, and it would make a lot of even tangentially related things more difficult.

I was thinking about how specifically it should be done, and at very least it would involve separating out the ffi part or using an existing ffi crate, sure, but it would also involve probably allowing the lua library to be built outside of the crate.  This comes with its own set of problems, like not being able to necessarily predict what all the types are going to be, and having to probably run bindgen on the header.  Maybe I could get away with having a lua53-raw and luaJIT51-raw, and have both build lua internally?

Argh, this is probably not something I will be doing soon at the very least.

**Comment by kyren** [comment]
If you don't mind, I'd like to close this issue though.  I mean this very genuinely though, if you know a smarter way to detect an incomplete statement, please please let me know what it is!

**Comment by calviken** [comment]
That makes sense. I don't use REPL much so that didn't really occur to me. 

As for LuaJIT, I'm beginning to think it would be better to focus on 5.3. The performance increase from LuaJIT really only make a difference when you're using it stand-alone (as opposed to creating an API). Plus as you already know, supporting multiple versions of anything is always going to be a huge pain.

Fact is, I got a little overexcited. This was actually my first attempt at contributing to open-source. Turns out it's not as easy as making terrible hacks for personal use.

Thanks for your time.
